File: library-app/src/main/java/com/yoon/libraryapp/domain/user/loanhistory/UserLoanHistoryRepository.java
package com.yoon.libraryapp.domain.user.loanhistory;

import com.yoon.libraryapp.domain.user.loanHistory.UserLoanHistory;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserLoanHistoryRepository extends JpaRepository<UserLoanHistory, Long> {

  UserLoanHistory findByBookNameAndIsReturn(String bookName, boolean isReturn);

}

File: library-app/src/main/kotlin/com/yoon/libraryapp/domain/user/loanHistory/UserLoanHistoryRepository.kt
package com.yoon.libraryapp.domain.user.loanHistory

import org.springframework.data.jpa.repository.JpaRepository

interface UserLoanHistoryRepository : JpaRepository<UserLoanHistory, Long>{

    fun findByBookName(bookName: String) : UserLoanHistory?

    fun findByBookNameAndStatus(bookName: String, status: UserLoanStatus) : UserLoanHistory?

    fun countByStatus(status: UserLoanStatus) : Long
}